custom/_legacy/hsph/tasks.py

```diff
@@ -235,22 +220,27 @@ def new_update_case_properties():
                     "closed_status": "timed_out_lost_to_follow_up",
                     "next_assignment": '',
                 }
-                cases_to_modify.append(
-                    {
-                        "case_id": case._id,
-                        "update": update,
-                        "close": True,
-                    }
-                )
-        case_blocks = []
-        for case in cases_to_modify:
-            kwargs = {
-                "create": False,
-                "case_id": case["case_id"],
-                "update": case["update"],
-                "close": case["close"],
-            }
-            if case.get("owner_id", None):
-                kwargs["owner_id"] = case["owner_id"]
-            case_blocks.append(ElementTree.tostring(CaseBlock(**kwargs).as_xml()))
-        submit_case_blocks(case_blocks, domain)
+                yield {
+                    "case_id": case._id,
+                    "update": update,
+                    "close": True,
+                }, domain
+
+
+@periodic_task(
+    run_every=crontab(minute=1, hour="*/6"),
+    queue=getattr(settings, 'CELERY_PERIODIC_QUEUE', 'celery')
+)
+def new_update_case_properties():
+    for case, domain in get_cases_to_modify():
+        kwargs = {
+            'create': False,
+            'case_id': case['case_id'],
+            'update': case['update'],
+            'close': case['close'],
+        }
+        if case.get('owner_id', None):
+            kwargs['owner_id'] = case['owner_id']
+        case_block = ElementTree.tostring(CaseBlock(**kwargs).as_xml())
+        # FB 185783: Do one at a time
+        submit_case_blocks(case_block, domain)
```

Comment:
I'm curious whether it's actually necessary to do this bit one at a time, considering we're already using generators to fetch the cases, and to return those that need to be modified.